protocols/raft/src/main/java/io/atomix/protocols/raft/impl/RaftContext.java

```diff
@@ -725,7 +729,8 @@ private void registerHandlers(RaftServerProtocol protocol) {
     }
   }
 
-  private <R extends RaftResponse> CompletableFuture<R> runOnContext(Supplier<CompletableFuture<R>> function) {
+  private <R extends RaftResponse> CompletableFuture<R> runOnContext(String requestType, Supplier<CompletableFuture<R>> function) {
```

Comment:
Is `runOnContext` strictly use for receiving messages? What if instead we added the metrics in the `RaftMessagingProtocol#registerHandler`, where we register message handlers for a given message type?

Reply:
Yes. `runOnContext` is only used for receiving messages. It can be added in RaftMessagingProtocol. We need to write a wrapper handler for each of the message. I will give it a try.